File: app/src/main/java/com/mmdev/meetups/ui/card/CardStackAdapter.java
package com.mmdev.meetups.ui.card;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;
import android.widget.Toast;

import com.mmdev.meetups.R;
import com.mmdev.meetups.models.ProfileModel;
import com.mmdev.meetups.utils.GlideApp;

import java.util.List;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

public class CardStackAdapter extends RecyclerView.Adapter<CardStackAdapter.ViewHolder>
{

	private List<ProfileModel> mUsersList;

	CardStackAdapter (List<ProfileModel> mUsersList) { this.mUsersList = mUsersList; }

	@NonNull
	@Override
	public ViewHolder onCreateViewHolder (@NonNull ViewGroup parent, int viewType) {
		LayoutInflater inflater = LayoutInflater.from(parent.getContext());
		View itemView = inflater.inflate(R.layout.fragment_card_item, parent, false);
		return new ViewHolder(itemView);
	}

	@Override
	public void onBindViewHolder (@NonNull ViewHolder holder, final int position) {
		ProfileModel user = mUsersList.get(position);
		holder.tvNameCard.setText(user.getName());
		holder.tvCityCard.setText(user.getCity());
		GlideApp.with(holder.tvImageCard)
				.load(user.getMainPhotoUrl())
				.into(holder.tvImageCard);
		holder.itemView
				.setOnClickListener
						(v -> Toast.makeText(v.getContext(), "clicked", Toast.LENGTH_SHORT).show());

	}

	@Override
	public int getItemCount () { return mUsersList.size(); }

	@Override
	public long getItemId (int position) { return position; }

	ProfileModel getSwipeProfile (int position){ return mUsersList.get(position); }

	static class ViewHolder extends RecyclerView.ViewHolder
	{
		private final TextView tvNameCard;
		private final TextView tvCityCard;
		private final ImageView tvImageCard;

		private ViewHolder (@NonNull final View itemView) {
			super(itemView);
			tvNameCard = itemView.findViewById(R.id.fragment_card_item_text_name);
			tvCityCard = itemView.findViewById(R.id.fragment_card_item_text_city);
			tvImageCard = itemView.findViewById(R.id.fragment_card_item_img_photo);
		}

	}

}

File: app/src/main/java/com/mmdev/meetapp/ui/card/CardStackAdapter.kt
package com.mmdev.meetapp.ui.card

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import android.widget.Toast
import androidx.recyclerview.widget.RecyclerView
import com.mmdev.meetapp.R
import com.mmdev.meetapp.models.ProfileModel
import com.mmdev.meetapp.utils.GlideApp

class CardStackAdapter internal constructor(private val mUsersList: List<ProfileModel>):
		RecyclerView.Adapter<CardStackAdapter.ViewHolder>() {

	override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
		val itemView = LayoutInflater.from(parent.context).inflate(R.layout.fragment_card_item, parent, false)
		return ViewHolder(itemView)
	}

	override fun onBindViewHolder(holder: ViewHolder, position: Int) {
		val (name, city, _, _, mainPhotoUrl) = mUsersList[position]
		holder.tvNameCard.text = name
		holder.tvCityCard.text = city
		GlideApp.with(holder.tvImageCard).load(mainPhotoUrl).into(holder.tvImageCard)
		holder.itemView.setOnClickListener { v -> Toast.makeText(v.context, "clicked", Toast.LENGTH_SHORT).show() }

	}

	override fun getItemCount(): Int { return mUsersList.size }

	override fun getItemId(position: Int): Long { return position.toLong() }

	internal fun getSwipeProfile(position: Int): ProfileModel { return mUsersList[position] }

	inner class ViewHolder (itemView: View): RecyclerView.ViewHolder(itemView) {
		internal val tvNameCard: TextView = itemView.findViewById(R.id.fragment_card_item_text_name)
		internal val tvCityCard: TextView = itemView.findViewById(R.id.fragment_card_item_text_city)
		internal val tvImageCard: ImageView = itemView.findViewById(R.id.fragment_card_item_img_photo)

	}

}
